flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -590,6 +604,15 @@ public Execution resetForNewExecution(final long timestamp, final long originati
 
 				this.currentExecution = newExecution;
 
+				synchronized (this.inputSplits){
+					for (InputSplit split: this.inputSplits){
+						if (split != null) {
+							this.jobVertex.getSplitAssigner().returnInputSplit(split, this.getParallelSubtaskIndex());
+						}
+					}
+					this.inputSplits.clear();
```

Comment:
I think in general the code-style in Flink avoids prefixing members with `this.` outside of constructors, getters, and setters.

Reply:
Done.